Library/Homebrew/test/livecheck/livecheck_spec.rb

```diff
@@ -0,0 +1,157 @@
+# Frozen_string_literal: true
+
+require "livecheck/livecheck"
+
+describe Homebrew::Livecheck do
+  subject(:livecheck) { described_class }
+
+  let(:f) do
+    formula("test") do
+      desc "Test formula"
+      homepage "https://brew.sh"
+      url "https://brew.sh/test-0.0.1.tgz"
+      head "https://github.com/Homebrew/brew.git"
+
+      livecheck do
+        url(+"https://github.s3.amazonaws.com/Homebrew/brew/releases/latest")
```

Comment:
```suggestion
        url "https://github.s3.amazonaws.com/Homebrew/brew/releases/latest"
```

Maybe it's just my tired brain but is there a reason for the `url(+"...")` format here instead of `url "..."`?

Reply:
Ah we need the String to be unfrozen because our preprocess method modifies it inplace. Without the unary `+`, we'd have a frozen string (because of `# frozen_string_literal: true`), which would cause `preprocess_url` and `livecheck_formulae` to error.

Reply:
Might be worth doing a `dup` instead?

Reply:
`brew style` didn't agree with that and suggested that I change it to this.

Reply:
@nandahkrishna I'm suggesting it be done in `preprocess_url` and `livecheck_formulae` on the incoming parameter.

Reply:
Ah okay. Thought you meant here, haha 😅

Reply:
> Without the unary `+`, we'd have a frozen string (because of `# frozen_string_literal: true`), which would cause `preprocess_url` and `livecheck_formulae` to error.

Ah, we simply need to replace the `sub!` in `preprocess_url` with `url = url.sub(...` and then this works fine with the usual `url "..."` format. I'll add a suggestion for the relevant line.

Reply:
> Ah, we simply need to replace the `sub!` in `preprocess_url` with `url = url.sub(...` and then this works fine with the usual `url "..."` format. I'll add a suggestion for the relevant line.

Sounds good!